src/main/java/seedu/address/logic/commands/DeleteCustomerCommand.java

```diff
@@ -37,37 +38,50 @@ public DeleteCustomerCommand(Phone targetPhone) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Customer> lastShownList = model.getFilteredCustomerList();
-        List<Order> orderList = model.getFilteredOrderList();
-        model.updateFilteredOrderList(PREDICATE_SHOW_ALL_ORDERS);
 
-        Index targetIndex = Index.fromZeroBased(lastShownList.size() + 1);
+        Customer customerToDelete = getCustomerToDelete(model);
+        cascadeDeleteOrders(model, customerToDelete);
+        model.deleteCustomer(customerToDelete);
+        model.setPanelToCustomerList(); // Display customer list
+
+        return new CommandResult(String.format(MESSAGE_DELETE_CUSTOMER_SUCCESS, customerToDelete));
+    }
+
+    /**
+     * Searches for the {@code Customer} to delete using the target phone number.
+     */
+    private Customer getCustomerToDelete(Model model) throws CommandException {
+        model.updateFilteredCustomerList(PREDICATE_SHOW_ALL_CUSTOMERS);
```

Comment:
Would suggest to use the `getCustomerWithPhone` method in `AddressBook` instead of filtering the whole list